.gitattributes

```diff
@@ -0,0 +1 @@
+poetry.lock binary
```

Comment:
What does this do?

Reply:
This basically makes diffs easier to deal with. It's not absolutely necessary, and I'd be fine removing it, but diffs of this file are essentially useless since it is entirely machine generated.

The main thing this does is to tell git to not even bother trying to show a diff.

Reply:
I get that... but then the counter-argument is that if somebody updates dependencies but forgets to update the lock file, I want to see on the PR that they haven't modified the lock file. Perhaps we can add a pre-commit check that the lock file is unchanged after an install or something? In cases with lock files, I don't actually care what the changes _are_, only that there _is_ a change when someone has modified dependencies.